src/lsm/compaction.zig

```diff
@@ -105,6 +105,7 @@ pub fn CompactionType(
 
         // Passed by `init`.
         tree_name: []const u8,
+        tree_id: u128,
```

Comment:
Suggestion: instead of adding the fields `tree_name` and `tree_id` we could reference `tree_config: Tree.Config` (or even better `*const Tree.Config` if possible).

IMO it reads better and decouples these fields from the compaction logic.